### PR TITLE
Fixes for undoing node actions from code and vice versa

### PR DESCRIPTION
**Pull request by mikusp**
### Pull Request Description

Replaces default atom behaviour of undoing changes in code by sending whole file as Substitute request and unifies it with out undo/redo infrastructure.

### Important Notes

@pmlodawski @ljkania I have replaced generating client UUID with zeros, because this caused accidental breakage in undo - node editor was considered to be different client than code editor. Ideally, this ID should be generated before and used in these two places. Please comment if you see a way to do that

### Checklist

- [ ] The documentation has been updated if necessary.
- [ ] The code conforms to the [Luna Style Guide](https://github.com/luna/wiki/blob/master/code-style/01.general.md) if it is Haskell.
- [ ] The code has been tested where possible.

### CODE REVIEW

File: luna-studio/node-editor/src/Main.hs [hunk 1]
```diff
@@ -21,14 +22,14 @@ import           WebSocket                  (WebSocket)
 runApp :: Chan (IO ()) -> WebSocket -> IO ()
 runApp chan socket = do
     random         <- newStdGen
-    clientId       <- generateUUID
+    -- clientId       <- generateUUID
```
Comment (pmlodawski) [inline luna-studio/node-editor/src/Main.hs]: Why?

File: luna-studio/text-editor/src/Main.hs [hunk 2]
```diff
@@ -17,11 +18,11 @@ import           WebSocket               (WebSocket)
 runApp :: Chan (IO ()) -> WebSocket -> IO ()
 runApp chan socket = do
     random       <- newStdGen
-    clientId             <- generateUUID
+    -- clientId             <- generateUUID
```
Comment (pmlodawski) [inline luna-studio/text-editor/src/Main.hs]: Remove it if its not needed

File: luna-studio/node-editor/src/Main.hs (hunk 1 above)
Comment (mikusp) [reply luna-studio/node-editor/src/Main.hs]: As I said in the description, now code editor and node editor have different clientIds, so undo treats them as different users. That means that if you change code, you cannot undo that in node editor, as instead your last action in node editor will be undone. This is confusing for users.
If you see a way to generate one ID for both code and node editors, I'm all for it.

Comment (pmlodawski) [reply luna-studio/node-editor/src/Main.hs]: I'm going to merge text-editor and code-editor in the future, can you hardcode single uuid in some common library and use in both this components?

Comment (mikusp) [reply luna-studio/node-editor/src/Main.hs]: Sure :+1:

**Pull request merged**